examples/simple-email-proof/vlayer/src/shared/hooks/useEmailProofVerification.ts

```diff
@@ -53,12 +53,15 @@ export const useEmailProofVerification = () => {
     hash: txHash,
   });
 
-  const { chain, error: chainError } = useChain(
+  const { chain, error: chainError } = useSyncChain(
     import.meta.env.VITE_CHAIN_NAME,
   );
-  if (chainError) {
-    throw new UseChainError(chainError);
-  }
+
+  useEffect(() => {
+    if (chainError) {
+      throw new ChainSyncError(chainError.message);
+    }
+  }, [chainError?.message]);
```

Comment:
Can you explain why it is required to wrap error handling in `useEffect`?

Reply:
error should be throw once. Imagine that state of component change and render function rerun should error be rethrown ?